plugins/techdocs/src/client.test.ts

```diff
@@ -35,27 +35,26 @@ const mockEntity = {
 
 describe('TechDocsStorageClient', () => {
   const mockBaseUrl = 'http://backstage:9191/api/techdocs';
-  const configApi = {
-    getOptionalString: () => 'http://backstage:9191/api/techdocs',
-  } as Partial<Config>;
+  const configApi = new MockConfigApi({
+    techdocs: { requestUrl: 'http://backstage:9191/api/techdocs' },
+  });
   const discoveryApi = UrlPatternDiscovery.compile(mockBaseUrl);
   const identityApi: jest.Mocked<IdentityApi> = {
-    getIdToken: jest.fn(),
-    getProfile: jest.fn(),
-    getUserId: jest.fn(),
-    signOut: jest.fn(),
-    getProfileInfo: jest.fn(),
-    getBackstageIdentity: jest.fn(),
     getCredentials: jest.fn(),
-  };
+  } as unknown as jest.Mocked<IdentityApi>;
+  const fetchApi = new MockFetchApi().setAuthorization({ identityApi });
```

Comment:
What do you think about `new MockFetchApi({ authorization: { identityApi }})` instead? (and move `implementation` in there too)

Reply:
Sure. I was pondering these a bit and since these features in turn use _other_ utility APIs (optionally!) it becomes a bit muddled. If middleware feature A and B both use the config api, shall we pass it in twice? :) better then perhaps to just have the deps as a flat set of root keys instead. So I'd suggest then something more like `new MockFetchApi({ authorization: { type: 'identity' }, identityApi }})` or so, and just throwing immediately if some needed input is missing.

This makes me think that if we get the DI system into place, then it could be really nice here too. Setting up a little container for all of this instead of being so explicit might have been nicer.

Reply:
Made a single-constructor version instead as suggsted.